test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "failed to install vote-bot", "failed to install vote-bot: %s\n%s", err, o)
+			}
+
+			out, err := TestHelper.KubectlApplyWithContext("", contexts["tgt"], "-f", "testdata/emojivoto-no-bot.yml")
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "failed to install emojivoto", "failed to install emojivoto: %s\n%s", err, out)
+			}
+
+		})
+
+		t.Run("Wait until target workloads are ready", func(t *testing.T) {
+			// Wait until client is up and running in source cluster
+			voteBotDeployReplica := map[string]testutil.DeploySpec{"vote-bot": {Namespace: ns, Replicas: 1}}
+			TestHelper.WaitRolloutWithContext(t, voteBotDeployReplica, contexts["src"])
+
+			// Wait until "target" services and replicas are up and running.
+			emojiDeployReplicas := map[string]testutil.DeploySpec{
+				"web":    {Namespace: ns, Replicas: 1},
+				"emoji":  {Namespace: ns, Replicas: 1},
+				"voting": {Namespace: ns, Replicas: 1},
+			}
+			TestHelper.WaitRolloutWithContext(t, emojiDeployReplicas, targetCtx)
+		})
+
+		timeout := time.Minute
+		err := TestHelper.RetryFor(timeout, func() error {
+			out, err := TestHelper.KubectlWithContext("",
+				targetCtx,
+				"--namespace", ns,
+				"logs",
+				"--selector", "app=web-svc",
+				"--container", "web-svc",
+			)
+			if err != nil {
+				return fmt.Errorf("%w\n%s", err, out)
+			}
+			// Check for expected error messages
+			for _, row := range strings.Split(out, "\n") {
+				if strings.Contains(row, " /api/vote?choice=:doughnut: ") {
+					return nil
+				}
+			}
+			return fmt.Errorf("web-svc logs in target cluster do not include voting errors\n%s", out)
+		})
+		if err != nil {
+			testutil.AnnotatedFatal(t, fmt.Sprintf("'linkerd multicluster gateways' command timed-out (%s)", timeout), err)
+		}
+	})
+}
+
+// TestMulticlusterStatefulSetTargetTraffic will test that a statefulset can be
+// mirrored from a target cluster to a source cluster. The test deploys two
+// workloads: a slow cooker (as a client) in the src, and an nginx statefulset in
+// (as a server) in the tgt. The slow-cooker is configured to send traffic to an
+// nginx endpoint mirror (nginx-statefulset-0). The traffic should be received
+// by the nginx pod in the tgt. To assert this, we get proxy metrics from the
+// gateway to make sure our connections from the source cluster were routed
+// correctly.
+func TestMulticlusterStatefulSetTargetTraffic(t *testing.T) {
+	if err := TestHelper.SwitchContext(contexts["tgt"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["tgt"], err)
+	}
+
+	ctx := context.Background()
+	// Create 'multicluster-statefulset' namespace in target cluster, to be deleted at the end of the test.
+	TestHelper.WithDataPlaneNamespace(ctx, "multicluster-statefulset", map[string]string{}, t, func(t *testing.T, ns string) {
```

Comment:
Same question as above here: we create a test namespace but then the manifests _also_ create their own namespaces.